src/nvim/ex_cmds2.c

```diff
@@ -3403,16 +3413,20 @@ static void script_host_execute_file(char *name, exarg_T *eap)
   // filename
   list_append_string(args, buffer, -1);
   // current range
-  list_append_number(args, eap->line1);
-  list_append_number(args, eap->line2);
+  assert(eap->line1 <= INT_MAX);
+  assert(eap->line2 <= INT_MAX);
+  list_append_number(args, (int)eap->line1);
+  list_append_number(args, (int)eap->line2);
```

Comment:
Likely just because vim does it, and noone has seriously suggested changing it :) Note as vim (and neovim) compiles works just fine on ILP32 or IL32P64 (64-bit windows) it cannot be that vim relies on `linenr_T > INT_MAX` as sentinel values or say temporary overflow in the midst of a calculation.
I don't like adding an assert for every technicallity, it makes sense on some casts but here it just adds noise. Rather we should continue to slowly but steadly refactor to use better and more consistent int types, to reduce the number of casts needed. (It might be worthwhile to try change `linenr_T` to say `int32_t` in the short run but it could generate warnings and follow-on changes, so it might not be an obvious change, but could be worth trying.)